app/presenters/contact_presenter.rb

```diff
@@ -52,9 +52,9 @@ def last_update
 
   private
 
-  def unmask_name?    
-    whitelisted_user? || ((whois_record.registrant.legal_person? ||
-      contact.attribute_disclosed?(:name)) && (captcha_solved? || registrant_publishable?))
+  def unmask_name?
+    whitelisted_user? || (whois_record.registrant.legal_person? &&
+      (captcha_solved? || registrant_publishable?)) || contact.attribute_disclosed?(:name)
```

Comment:
A very confusing construction, in which case it is better to put each expression into a variable for readability, as Uncle Bob did in Clean Code
```
publication_availability = captcha_solved? || registrant_publishable?
available_contact = whois_record.registrant.legal_person? && publication_availability
whitelisted_user? || available_contact || contact.attribute_disclosed?(:name)
```